src/sagemaker_xgboost_container/checkpointing.py

```diff
@@ -388,4 +388,16 @@ def callback(self, env):
             logger.debug("Not master (rank = %d). Exiting intermediate model callback.", env.rank)
             return
 
-        self._save_intermediate_model(env.model)
+        self.save_intermediate_model(env.model)
+
+
+class SaveIntermediateModelCallBack(xgb.callback.TrainingCallback):
+    """The new implementation of callback functions from 1.3."""
+    def __init__(self, intermediate_model_dir, model_name, is_master):
+        self.callback = SaveIntermediateModel(intermediate_model_dir, model_name)
+        self.is_master = is_master
+
+    def after_iteration(self, model, epoch, evals_log) -> bool:
+        if self.is_master:
+            self.callback.save_intermediate_model(model)
+        return False
```

Comment:
Is it right for this to always return False?

Reply:
Yes. If we turn True, it means to stop the training. Most callbacks always return False.

Reply:
early_stopping_data is a function parameter here. Although we always set the value in the caller, it is still better to do the sanity check here.